src/services/logDataService/searchQueryDataService.ts

```diff
@@ -0,0 +1,99 @@
+
+
+import { getBackendSrv } from '@grafana/runtime';
+import { lastValueFrom } from 'rxjs';
+
+import { ISearchAggregateResult, ISearchQueryDataService, ISearchStatus, ISearchCreate, SearchQueryParams } from './types';
+
+export class SearchQueryDataService implements ISearchQueryDataService {
+
+  create(
+    searchQueryParams : Omit<SearchQueryParams , 'paginationInfo' | 'types' >  
+  ): Promise<ISearchCreate> {
+
+    const { query , startTime , endTime  , baseUrl , basicAuth } = searchQueryParams
+    const data = {
+      query,
+      from : startTime,
+      to : endTime,
+      byReceiptTime: true
+
+    };
+    return  lastValueFrom(getBackendSrv().fetch<ISearchCreate>({
+      method : 'POST',
+      credentials: 'include',
+      headers: {
+        Authorization: basicAuth,
+      },
+      url: baseUrl + '/v1/search/jobs',
+      data,
+    })).then(response => response.data);
+  }
+
+  status(searchQuerySessionId: string , baseUrl : string , basicAuth:string): Promise<ISearchStatus> {
+    return lastValueFrom(getBackendSrv().fetch<ISearchStatus>({
+      method : 'GET',
+      credentials: 'include',
+      headers: {
+        Authorization: basicAuth,
+      },
+      url: baseUrl + '/v1/search/jobs/' + searchQuerySessionId,
+    })).then(response => response.data)
+  }
+
+  delete(searchQuerySessionId: string , baseUrl : string , basicAuth  :string): Promise<any>{
+    return lastValueFrom(getBackendSrv().fetch<ISearchStatus>({
+      method : 'DELETE',
+      credentials: 'include',
+      headers: {
+        Authorization: basicAuth,
+      },
+      url: baseUrl + '/v1/search/jobs/' + searchQuerySessionId,
```

Comment:
Maybe a constant for '/v1/search/jobs/' or a util method to get the url with session id
```suggestion
      url: baseUrl + SEARCH_JOB_BASE_URL + searchQuerySessionId,
```